### PR TITLE
Clarify terms

### PR DESCRIPTION
**Pull request by richlander**
Update to clarify terms for packages, due to questions asked at https://github.com/dotnet/core/issues/9658.

### CODE REVIEW

File: license-information.md [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
 
 ## Binaries
 
-Project binaries (like the .NET runtime distribution) primarily use the MIT license. Binaries built for macOS and Linux exclusively use the MIT license.
+Microsoft built binaries use the MIT license, for:
+
+- NuGet packages
+- Runtime (including ASP.NET Core) and SDK builds for Linux and macOS
+
+The .NET runtime [statically links some binaries](https://github.com/dotnet/runtime/tree/main/src/native/external), all of which are compatible with the specified runtime license.
```
Comment (jkotas) [inline license-information.md:27]: This is incomplete information that is likely raise more questions and requests for clarifications. There is other 3rd party stuff that gets statically linked into Microsoft built binaries - VC++ runtime, parts of C++ standard library on macOS/Linux, https://github.com/dotnet/icu, ... . I think THIRD-PARTY-NOTICES.TXT mention above is sufficient.

Comment (richlander) [reply license-information.md:27]:
```suggestion
```

Comment (richlander) [reply license-information.md:27]: Feel free to commit and merge if you are happy with the rest.

File: license-information.md [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
+Third party notice files are used to document source copied (and possibly further modified) from other projects.
 
 [Project copyright guidance](https://github.com/dotnet/runtime/blob/main/docs/project/copyright.md) provides more details on our policies.
 
 ## Binaries
 
-Project binaries (like the .NET runtime distribution) primarily use the MIT license. Binaries built for macOS and Linux exclusively use the MIT license.
+Microsoft built binaries use the MIT license, for:
+
+- NuGet packages
```
Comment (jkotas) [inline license-information.md:24]: Runtime is delivered via NuGet packages too, so this description is ambiguous.

https://github.com/dotnet/runtime/blob/main/docs/project/licensing-assets.md solves this problem by talking about product distributions and package distributions. Should we do the same here?

File: license-information.md [hunk 3]
```diff
@@ -1,29 +1,41 @@
...
 
-Parts of the .NET runtime are embedded in applications, including [platform-specific executable hosts](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#framework-dependent-executable), and [self-contained deployments](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#self-contained-deployment), are subject to [.NET](https://github.com/dotnet/dotnet/blob/main/LICENSE.TXT) and [third-party notice](https://github.com/dotnet/dotnet/blob/main/THIRD-PARTY-NOTICES.txt) license terms.
+Parts of the .NET runtime are embedded in applications, including [platform-specific executable hosts](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#framework-dependent-executable),
+and [self-contained deployments](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#self-contained-deployment), are subject to [.NET](https://github.com/dotnet/dotnet/blob/main/LICENSE.TXT) and [third-party notice](https://github.com/dotnet/dotnet/blob/main/THIRD-PARTY-NOTICES.txt) license terms.
 
 Binaries that target Windows are subject to [additional terms](license-information-windows.md).
```
Comment (jkotas) [inline license-information.md:41]: This applies to runtime only. It does not apply to NuGet packages with Windows-specific libraries, like `Microsoft.Win32.Registry`.

File: license-information.md (hunk 2 above)
Comment (richlander) [reply license-information.md:24]: Yes. I paused at that when I wrote it. I'll fix it.

File: license-information.md [hunk 4]
```diff
@@ -6,36 +6,40 @@ This document is provided for informative purposes only and is not itself a lice
 
 ## Source code
 
-.NET source uses the MIT license. There are several [project repos](./Documentation/core-repos.md).
+.NET source uses the MIT license.
 
-Each repo has:
+[Each repo](./Documentation/core-repos.md)) has:
```
Comment (jkotas) [inline license-information.md:11]:
```suggestion
[Each repo](./Documentation/core-repos.md) has:

```
Typo

**Review by jkotas — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**